services/arrow-flight/src/test/java/com/dremio/service/flight/protector/TestCancellableUserResponseHandler.java

```diff
@@ -141,17 +132,8 @@ private void testClientCancelCaughtAndPropagatedToServer(Supplier<Boolean> isCan
     thrown.expect(FlightRuntimeException.class);
 
     final CancellableUserResponseHandler<BigDecimal> cancellableUserResponseHandler =
-      new CancellableUserResponseHandler<BigDecimal>(externalId, userSession, mockedUserWorkerProvider, isCancelled) {
-        @Override
-        public void sendData(RpcOutcomeListener<GeneralRPCProtos.Ack> outcomeListener, QueryWritableBatch result) {
-          fail();
-        }
-
-        @Override
-        public void completed(UserResult result) {
-          fail();
-        }
-      };
+      new CancellableUserResponseHandler<>(externalId, userSession, mockedUserWorkerProvider, isCancelled,
```

Comment:
Doesn't this test no longer fail correctly now that sendData()/completed() calls no longer call fail? Suggest we remove the final marker and leave those overrides in place.

Reply:
Not actually, the original test also didn't call sendData() or completed(), what happens if that `CancellableUserResponseHandler#get()` throws the exception that the test is expecting when it gets here: https://github.com/s1mbi0se/dremio-oss/blob/master/services/arrow-flight/src/main/java/com/dremio/service/flight/protector/CancellableUserResponseHandler.java#L83